hr_job_category/models/hr.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class HRJob(models.Model):
+    _inherit = "hr.job"
+
+    category_ids = fields.Many2many(
+        "hr.employee.category",
+        "job_category_rel",
+        "job_id",
+        "category_id",
+        string="Associated Tags",
+    )
+
+
+class HRContract(models.Model):
+    _inherit = "hr.contract"
+
+    def _remove_tags(self, job_id):
+        if isinstance(job_id, int):
+            job = self.env["hr.job"].browse(job_id)
+        else:
+            job = job_id
+        for employee in self.mapped("employee_id"):
+            _logger.debug(
+                "Removing employee tags if tags exist on contract job: %s",
+                employee.category_ids,
+            )
+            tags_to_remove = [
+                (3, tag.id) for tag in job.category_ids & employee.category_ids
+            ]
+            if tags_to_remove:
+                employee.write({"category_ids": tags_to_remove})
+
+    def _tag_employees(self, job_id):
+        if isinstance(job_id, int):
+            job = self.env["hr.job"].browse(job_id)
+        else:
+            job = job_id
+        _logger.debug(
+            "Adding employee tags if job tags doesn't exist: %s", job.category_ids
+        )
+        for employee in self.mapped("employee_id"):
+            tags_to_add = [
+                (4, tag.id) for tag in job.category_ids - employee.category_ids
+            ]
+            if tags_to_add:
+                employee.write({"category_ids": tags_to_add})
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
Since the create method in version 18 receives a list of vals as a parameter, it must be iterated to validate all possible values for each record. Therefore, the method must be written as follows, otherwise it won't work.
```suggestion
    def create(self, vals_list):
     res = super().create(vals_list)
        for vals in vals_list:
            if "job_id" in vals:
                res._tag_employees(vals.get("job_id"))
        return res
```